circom-prover/src/witness.rs

```diff
@@ -1,30 +1,30 @@
 use num::{BigInt, BigUint};
-use std::{collections::HashMap, str::FromStr, thread::JoinHandle};
+use std::{collections::HashMap, thread::JoinHandle};
+
+#[cfg(feature = "rustwitness")]
+use std::str::FromStr;
 
 #[cfg(feature = "witnesscalc")]
 use witnesscalc_adapter::parse_witness_to_bigints;
 
 /// Witness function signature for rust_witness (inputs) -> witness
-#[cfg(feature = "rustwitness")]
 type RustWitnessWtnsFn = fn(HashMap<String, Vec<BigInt>>) -> Vec<BigInt>;
 /// Witness function signature for witnesscalc_adapter (inputs) -> witness
-#[cfg(feature = "witnesscalc")]
 type WitnesscalcWtnsFn = fn(&str) -> anyhow::Result<Vec<u8>>;
 
 #[derive(Debug, Clone, Copy)]
 pub enum WitnessFn {
-    #[cfg(feature = "witnesscalc")]
```

Comment:
If we removed this feature config, it could cause an unexpected error. `generate_witness` might return an empty array in L49 which causes proving error. And it's a runtime error, harder to troublshooting. Is it your intention?